File: 2315-count-asterisks/2315-count-asterisks.py
class Solution:
    def countAsterisks(self, s: str) -> int:
        count=0
        inside_pair=False
        for i in s:
            if i=='|':
                inside_pair=not inside_pair
                
            elif i=='*':
                if not inside_pair:
                    count+=1
        return count    
            

File: 2315-count-asterisks/2315-count-asterisks.java
class Solution {
    public int countAsterisks(String s) {
        int c=0;
        boolean inside=false;
        for(int i=0;i<s.length();i++){
            if(s.charAt(i)=='|'){
                inside=!inside;
            }
            else if(!inside && s.charAt(i)=='*'){
                c+=1;
            }
        }
        return c;
    }
}